lib/round.rb

```diff
@@ -0,0 +1,96 @@
+require './lib/deck'
+
+class Round
+  attr_reader :deck,
+              :turns
+  attr_accessor :correct_answers,
+                :turn_count,
+                :total_cards,
+                :categories
+
+  def initialize(deck)
+    @deck = deck
+    @turns = []
+    @correct_answers = 0
+    @turn_count = 1
+    @total_cards = 0
+    @categories = []
+  end
+
+  def current_card
+    @deck.cards.first
+  end
+
+  def take_turn(answer, card = current_card)
+    @deck.cards.shift
+    @turn_count += 1
+    new_turn = Turn.new(answer, card)
+    @turns << new_turn
+    @correct_answers += 1 if @turns.last.correct?
+    new_turn
+  end
+
+  def number_correct
+    @correct_answers
+  end
+
+  def percent_correct
+    (@correct_answers.to_f / turns.count) * 100
+  end
+
+  def number_correct_by_category(cat)
+    num_by_cat = turns.count do |turn|
+      turn.correct? && turn.card.category == cat
+    end
+  end
+
+  def percent_correct_by_category(cat)
+    total_by_cat = @turns.count do |turn|
+      turn.card.category == cat
+    end
+    (number_correct_by_category(cat).to_f / total_by_cat) * 100
+  end
+
+  # this seems like an ideal place to use #select, but
+  # @categories = deck.cards.select do |card|
+  #   card.category if @categories.include?(card.category)
+  # end
+  # doesn't pass tests for method
+  def collect_category
+    deck.cards.each do |card|
+      @categories << card.category if @categories.include?(card.category) == false
+    end
+  end
+
+  def start
+    @total_cards = deck.cards.count
+    collect_category
```

Comment:
This functionality doesn't feel like it belongs within the responsibility of this method. I wonder if the logic for collect_categories could make more sense in the Deck class instead of the Round class.